controller/volume_controller.go

```diff
@@ -809,6 +812,9 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, e *longhorn
 			return nil
 		}
 
+		//Restore the replicas if required
+		vc.Restore(v, e)
```

Comment:
It's better to restore in the engine controller instead of volume controller. Engine controller is polling the engine every 5 seconds, so you can initiate the restoring if it's not already happening, then check for the status in the async way.

Reply:
Unaddressed.